rdrf/rdrf/views/context_views.py

```diff
@@ -59,48 +54,39 @@ def get_context_launcher(self, user, registry_model, patient_model, context_mode
         return context_launcher.html
 
     def get_naming_info(self, form_group_id):
-        if form_group_id is not None:
-            context_form_group = ContextFormGroup.objects.get(id=form_group_id)
-            return context_form_group.naming_info
-        else:
+        if form_group_id is None:
             return "Display Name will default to 'Modules' if left blank"
+        context_form_group = ContextFormGroup.objects.get(id=form_group_id)
+        return context_form_group.naming_info
 
     def get_default_name(self, patient_model, context_form_group_model):
         if context_form_group_model is None:
             return "Modules"
-        else:
-            return context_form_group_model.get_default_name(patient_model)
+        return context_form_group_model.get_default_name(patient_model)
 
-    def allowed(self, user, registry_code, patient_id, context_id):
+    def allowed(self, user, registry_code, patient_id, context_id=None):
```

Comment:
OK, I'll bite :-).
Why the change? When can the `context_id` be `None`?
And how wouldn't the code at line `81` fail if `None` is received?
Is that failure masked by the catch-all `try/except`? If that's the intent I would rather make it explicit ie. short-cut the whole processing with an `if context_id is None: return False` at the beginning of the method.

Reply:
context_id can be none in case a new context gets created. Line 81 was changed to only do the check if the context exists.